### DISCUSSION TITLE
L2 ERC-20 contract is not created by the deposit transaction

### DISCUSSION THREAD

**Discussion by elek** [zkSync-Community-Hub/zksync-developers#159] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Bridge ERC-20 tokens.

### What is the specific issue or error you're encountering?

After calling the `deposit` method on L12L2 bridge, the transaction is successful, but L2 contract is not created.

Example transaction on goerli: https://etherscan.io/tx/0x33c81aea51b66b4fa3b89503d626d499b4a3ec85b349bd987483eca45af00983

On main: https://etherscan.io/tx/0x33c81aea51b66b4fa3b89503d626d499b4a3ec85b349bd987483eca45af00983

Earlier it worked, for example here (but with 0 value!!!):

https://goerli.etherscan.io/tx/0x2135f1e9c5cb44d090e23e8e17d14b4ce494656f25af38c4a91823ad911fcf7e

But now the similar transaction fails with the following error:

```
2023/10/02 12:31:05 execution reverted: mv
```

Not full clear why, but looks like I need to set value. When I set, the transaction is finished but balance is not set / bridged ERC-20 contract is not deployed.

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

This is the followup of this discussion: https://github.com/zkSync-Community-Hub/zkync-developers/discussions/121#discussioncomment-6923964

The javascript library is fixed as a result of that question, but this part of the discussion is not answered.

I couldn't use js library, as it's not released yet + the fix is committed to an ether6 feature branch.

**Comment by idea404** [comment, marked as answer]
Hi @elek, bridging ERC20 tokens is now possible with the ethers5 version of `zsync-web3` JS SDK. Dustin also [added a working script you can use](https://github.com/zkSync-Community-Hub/zkync-developers/discussions/121#discussioncomment-7187842) for the purpose. You can install the ethers5 version of the package as opposed to ethers by installing it as `npm i zksync-web3@0.15.5`.

**Reply by elek** [reply]
Thanks the answer, yes, the working script helped a lot (even if it should be restarted as hanging between the two transactions).

But still I don't understand why this transaction didn't create the ERC-20 bridge.

We would prefer to create similar transactions from our toolset, but I have concern about the safety of sending such transactions.

I assume there is some error in the input parameters, but I have no idea what was it and I couldn't see any error message anywhere.

Still I would like to know what was my mistake with the transaction, and and what was missing to create the bridge contract.

**Reply by idea404** [reply]
Hey @elek in this case best you can do is add more logging and look closer to the contracts involved in this bridge transaction. The implementation of `wallet.zkWallet.deposit()` should be performing the bridging function calling the relevant contracts you can query for more details

**Comment by rietracerdanchri** [comment]
It appears that the issue may be related to the deployment of the ERC-20 contract on Layer 2 not being triggered successfully after the deposit transaction. Ensure that the deposit transaction includes all necessary parameters and data required for the deployment of the ERC-20 contract on Layer 2. Double-check the configuration and parameters of the deposit method to ensure that it aligns with the requirements of the bridge protocol. Additionally, review any recent changes to the contract or deployment process that may have affected the deployment mechanism.

**Comment by neumagbutorcapri** [comment]
Ensure that the value sent with the deposit transaction matches the expected value specified by the contract. If the contract expects a non-zero value, make sure to provide it when calling the deposit method.

**Comment by singletonbrittany** [comment]
It seems like there might be an issue with the transaction execution, potentially related to the value being transferred. Ensure that the correct value is being set when calling the deposit method on the L12L2 bridge contract. Additionally, verify that the necessary permissions and configurations are properly set up for deploying the bridged ERC-20 contract on the L2 network. Double-checking these parameters could help resolve the problem.

**Comment by swelexdewi** [comment]
The user is encountering issues with creating an L2 contract after calling the deposit method on the L12L2 bridge in the Testnet environment. They have provided transaction examples and error messages, and suspect that setting a value may be necessary. The user has not made recent changes to the contract and is not using external libraries or contracts. They have tried to resolve the issue by discussing it in a previous thread and fixing the JavaScript library. However, they are still facing issues with the contract and are seeking further assistance.

**Comment by sanddethe** [comment]
It seems that after calling the deposit method on the L12L2 bridge, the transaction is successful, but the L2 contract is not created. The error message received is "execution reverted: mv". It is unclear why this is happening, but it may be necessary to set a value. When the value is set, the transaction is completed, but the balance is not set and the bridged ERC-20 contract is not deployed. It is suggested to use the JavaScript library once it's released or wait for the fix to be committed to an ether6 feature branch.
